### PR TITLE
DT-292: added telemetryModule to add client_id and user_name as custo…

### PR DESCRIPTION
**Pull request by SimonMitchellMOJ**
…mDimension to app insights

### CODE REVIEW

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/config/ClientTrackingTelemetryModuleTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    @Autowired
+    private ClientTrackingTelemetryModule clientTrackingTelemetryModule;
+
+    @Autowired
+    private JwtAuthHelper jwtAuthHelper;
+
+    @Before
+    public void setup() {
+        ThreadContext.setRequestTelemetryContext(new RequestTelemetryContext(1L));
```
Comment (andrewrlee) [inline src/test/java/uk/gov/justice/digital/hmpps/oauth2server/config/ClientTrackingTelemetryModuleTest.java]: Is it worth tearing down afterwards?

File: build.gradle [hunk 2]
```diff
@@ -116,6 +116,8 @@ dependencies {
 
   compileOnly 'org.projectlombok:lombok:1.18.8'
 
+  testAnnotationProcessor 'org.projectlombok:lombok:1.18.8'
```
Comment (petergphillips) [inline build.gradle]: should really match the `implementation` version - i.e. 1.18.10

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/ClientTrackingTelemetryModule.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.oauth2.provider.token.store.KeyStoreKeyFactory;
+
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.security.KeyPair;
+
+@Configuration
+public class ClientTrackingTelemetryModule implements WebTelemetryModule, TelemetryModule {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/ClientTrackingTelemetryModule.java]: think you only need to implement `WebTelemetryModule`?  That's the only one you are implementing

Comment (SimonMitchellMOJ) [reply src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/ClientTrackingTelemetryModule.java]: TelemetryModule is required, if its not there the customDimensions are not added to app insights

**Pull request merged**